scripts/deploy/32_eth_autocall_vault.ts

```diff
@@ -0,0 +1,209 @@
+  OptionsPremiumPricerInStables_BYTECODE,
+  ETH_PRICE_ORACLE,
+  WETH_ADDRESS,
+  USDC_ADDRESS,
+  OTOKEN_FACTORY,
+  GAMMA_CONTROLLER,
+  MARGIN_POOL,
+} from "../../constants/constants";
+import OptionsPremiumPricerInStables_ABI from "../../constants/abis/OptionsPremiumPricerInStables.json";
+import ManualVolOracle_ABI from "../../constants/abis/ManualVolOracle.json";
+import { AUCTION_DURATION, PREMIUM_DISCOUNT } from "../utils/constants";
+
+import { getDeltaStep } from "../../test/helpers/utils";
+
+const main = async ({
+  network,
+  deployments,
+  ethers,
+  getNamedAccounts,
+}: HardhatRuntimeEnvironment) => {
+  const { BigNumber } = ethers;
+  const { parseEther } = ethers.utils;
+  const { deploy } = deployments;
+  const { deployer, owner, keeper, admin, feeRecipient } =
+    await getNamedAccounts();
+  console.log(`32 - Deploying ETH Autocall Vault on ${network.name}`);
+
+  const chainId = network.config.chainId;
+  if (chainId !== CHAINID.ETH_MAINNET) {
+    console.log(`Error: chainId ${chainId} not supported`);
+    return;
+  }
+
+  const manualVolOracle = await deployments.get("ManualVolOracle");
+  const underlyingOracle = ETH_PRICE_ORACLE[chainId];
+  const stablesOracle = USDC_PRICE_ORACLE[chainId];
+
+  const manualVolOracleContract = await ethers.getContractAt(
+    ManualVolOracle_ABI,
+    manualVolOracle.address
+  );
+  const optionId = await manualVolOracleContract.getOptionId(
+    getDeltaStep("WETH"),
+    WETH_ADDRESS[chainId],
+    USDC_ADDRESS[chainId],
+    true
+  );
+
+  const pricer = await deploy("OptionsPremiumPricerWETH", {
+    from: deployer,
+    contract: {
+      abi: OptionsPremiumPricerInStables_ABI,
+      bytecode: OptionsPremiumPricerInStables_BYTECODE,
+    },
+    args: [optionId, manualVolOracle.address, underlyingOracle, stablesOracle],
+  });
+
+  console.log(`AutocallVaultWETH pricer @ ${pricer.address}`);
+
+  // Can't verify pricer because it's compiled with 0.7.3
+
+  const strikeSelection = await deploy("ManualStrikeSelectionAutocall", {
+    contract: "ManualStrikeSelection",
+    from: deployer,
+    args: [],
+  });
+
+  console.log(`AutocallVaultWETH strikeSelection @ ${strikeSelection.address}`);
+
+  try {
+    await run("verify:verify", {
+      address: strikeSelection.address,
+      constructorArguments: [],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  const lifecycleTreasury = await deploy("VaultLifecycleTreasury", {
+    contract: "VaultLifecycleTreasury",
+    from: deployer,
+  });
+  console.log(`VaultLifeCycleTreasury @ ${lifecycleTreasury.address}`);
+
+  // deploy logic
+  const logicDeployment = await deploy("RibbonAutocallVault", {
+    contract: "RibbonAutocallVault",
+    from: deployer,
+    args: [
+      USDC_ADDRESS[chainId],
+      OTOKEN_FACTORY[chainId],
+      GAMMA_CONTROLLER[chainId],
+      MARGIN_POOL[chainId],
+    ],
+    libraries: {
+      VaultLifecycleTreasury: lifecycleTreasury.address,
+    },
+  });
+
```

Comment:
can use same options premium pricer we use for put-selling vault
can use new manual strike selection + lifecycle treasury

Reply:
https://github.com/ribbon-finance/ribbon-v2/pull/313/commits/7cf6f18f834bf6281f3be92a03e395c7a5038916